compiler/circle-opselector/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+set(DRIVER "driver/Driver.cpp")
+
+file(GLOB_RECURSE SOURCES "src/*.cpp")
+
+add_executable(circle-opselector ${DRIVER} ${SOURCES})
+target_include_directories(circle-opselector PRIVATE src)
+target_link_libraries(circle-opselector foder)
+target_link_libraries(circle-opselector safemain)
+target_link_libraries(circle-opselector loco)
+target_link_libraries(circle-opselector luci_import)
+target_link_libraries(circle-opselector luci_export)
+target_link_libraries(circle-opselector arser)
+target_link_libraries(circle-opselector vconone)
+target_link_libraries(circle-opselector luci_service)
+target_link_libraries(circle-opselector luci_profile)
+
+install(TARGETS circle-opselector DESTINATION bin)
```

Comment:
I don't see any unit test codes here and draft codes.
Unit test codes are important to check your codes are working as expected.
And our SE requires test coverage to be in good shape.

Reply:
@seanshpark Other two members from SSAFY team are working for test. The 2nd draft is in #7765, which will use `luci-interpreter` as you suggested. For your information, they've checked this code by local test. I've got how they did https://github.com/Samsung/ONE/pull/7600#issuecomment-930781949 and https://github.com/Samsung/ONE/pull/7600#issuecomment-930791627.

Reply:
It's `UINT TESTING` not integration testing.

Reply:
@seanshpark What kind of unit tests do you want to be included? For example, CLI argument parsing will tested in `arg_validity_test.sh` aforementioned PR (#7765).

Reply:
> I don't see any unit test codes here and draft codes. Unit test codes are important to check your codes are working as expected. And our SE requires test coverage to be in good shape.

Also, could you please point out the example of unit test under `compiler` so that the SSAFY members can refer? For example, what is the unit tests of `tf2tflite-value`? I guess there are some in https://github.com/Samsung/ONE/tree/master/compiler/tf2tflite-value-pb-test, but I am not sure what is the unit tests.

Reply:
@glistening I think unit test here is function-level testing. These kind of files have suffix `*.test.cpp`. Please see https://github.com/Samsung/ONE/tree/31e5b3a6ade428688a3e96b891500a1cc0c841b1/compiler/loco/src dir and subdirectory. Each test checks if function or metohd or class works well. Of course not all the compiler modules have unit test. Actually, it should have written but was postponed due to schedule.

Writing unit test bothers us but it has several advantages.
- It helps us to identify the smallest defects that might be present in the units before integration testing where it may take some time to find some small defects.
- It is kind of document for developers. Developers who want to learn about the functionality of a certain module can easily learn about it by reading its unit test.

Reply:
> @glistening I think unit test here is function-level testing. These kind of files have suffix `*.test.cpp`. Please see ..

I know what is `unit test`. I am asking what would be the unit test for the case of `circle-opselector`. It is not a module to be used for other modules. It is an app using several modules. I cannot find the unit test for `tf2tflite` or `circle2circle`. What you've pointed out is `luci` which definitely can have a well defined unit test since it provides public interfaces and it is supposed to be used for other apps.

Reply:
### Search unit test for this kind of tool

There is only a *.test.cpp under `circle2circle`.
```
$ find compiler/circle2circle -name "*.test.*"
compiler/circle2circle/src/Circle2Circle.test.cpp
```

which just run without argument and check the exit code.

<sub><b>Circle2Circle.test.cpp</b></sub>
```c++
#include "TestHelper.h"

#include <gtest/gtest.h>

TEST(Circle2CircleTest, NoArg_NEG)
{
  Argv<1> argv;
  argv.add("circle2circle");

  ::testing::internal::CaptureStdout();
  int result = entry(1, argv.argv());
  ASSERT_EQ(255, result);
}
```

I cannot find unit test from `tf2tflite*`.
```
$ find compiler/tf2tflite -name "*.test.*"
$ find compiler/tf2tflite-value-pb-test -name "*.test.*"
$ find compiler/tf2tfliteV2 -name "*.test.*"
$ find compiler/tf2tflite-value-pb-test -name "*.test.*"
```

@mhs4670go or @seanshpark Could you give a guide so that COSMOS member can refer?